src/test/java/org/jboss/arquillian/extension/jacoco/test/unit/ApplicationArchiveInstrumenterTest.java

```diff
@@ -0,0 +1,84 @@
+package org.jboss.arquillian.extension.jacoco.test.unit;
+
+
+import org.jboss.arquillian.core.api.Instance;
+import org.jboss.arquillian.extension.jacoco.client.ApplicationArchiveInstrumenter;
+import org.jboss.arquillian.extension.jacoco.client.JacocoConfiguration;
+import org.jboss.arquillian.test.spi.TestClass;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.EnterpriseArchive;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+
+import org.junit.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.logging.Formatter;
+import java.util.logging.Handler;
+import java.util.logging.Logger;
+import java.util.logging.SimpleFormatter;
+import java.util.logging.StreamHandler;
+
+import static junit.framework.TestCase.assertNotNull;
+import static junit.framework.TestCase.assertTrue;
+
+
+public class ApplicationArchiveInstrumenterTest {
+
+    @Test
+    public void should_not_break_process_archive_if_it_contains_path_as_ear_type_and_log_warning() throws NoSuchFieldException, IllegalAccessException {
+        JavaArchive javaArchive = ShrinkWrap.create(JavaArchive.class, "dri.jar").addClass(DummyInstance.class)
+                .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml");
+
+        EnterpriseArchive enterpriseArchive =  ShrinkWrap.create(EnterpriseArchive.class, "test.ear")
+                .addAsLibraries(javaArchive)
+                .addAsManifestResource(EmptyAsset.INSTANCE, "application.xml")
+                .addAsManifestResource(EmptyAsset.INSTANCE, "dir.ear/pom.properties");
+
+        ApplicationArchiveInstrumenter applicationArchiveInstrumenter = new ApplicationArchiveInstrumenter();
+        setDefaultValueIfConfigIsNull(applicationArchiveInstrumenter);
+
+        Logger logger = Logger.getLogger(ApplicationArchiveInstrumenter.class.getName());
```

Comment:
Do we really have to test logger here? it's just an added value and doesn't bring a lot of benefit from the pure functionality perspective. The core logic is manipulating the archive and removing signatures. Isn't it more important to test?

We probably should have at least two tests here - one with the correct archive and one with directories with suffixes matching java artifacts.